workflow/scripts/featurecounts_deseq2.R

```diff
@@ -0,0 +1,382 @@
+                    MAplotFile <- snakemake@output[["plot_FDR_1_perc_MA"]]  # AVI: added to create separate pdf files
+                    VolcanoPlotFile <- snakemake@output[["plot_FDR_1_perc_volcano"]]  # AVI: added to create separate pdf files
+                }
+                if (MIN_FDR == 0.05) {
+                    CompResultsFile <- snakemake@output[["FDR_5_perc_res"]]
+                    CompBEDFile <- snakemake@output[["FDR_5_perc_bed"]]
+                    MAplotFile <- snakemake@output[["plot_FDR_5_perc_MA"]]  # AVI: added to create separate pdf files
+                    VolcanoPlotFile <- snakemake@output[["plot_FDR_5_perc_volcano"]]  # AVI: added to create separate pdf files
+                }
+
+                write.table(pass.fdr.table, file=CompResultsFile, col.names=TRUE, row.names=FALSE, sep='\t', quote=FALSE)
+                write.table(pass.fdr.table[,c("Chr","Start","End","Geneid","log2FoldChange","Strand")], file=CompBEDFile, col.names=FALSE, row.names=FALSE, sep='\t', quote=FALSE)
+
+                ## MA PLOT & VOLCANO PLOT
+                pdf(file=MAplotFile,width=10,height=8)  # AVI: added to create separate pdf files
+                DESeq2::plotMA(comp.results, main=paste("MA plot FDR <= ",MIN_FDR,sep=""), ylim=c(-2,2),alpha=MIN_FDR)
+                dev.off()  # AVI: added to create separate pdf files
+
+                pdf(file=VolcanoPlotFile,width=10,height=8)  # AVI: added to create separate pdf files
+                plot(comp.table$log2FoldChange, -1*log10(comp.table$padj), col=ifelse(comp.table$padj<=MIN_FDR, "red", "black"), xlab="logFC", ylab="-1*log10(FDR)", main=paste("Volcano plot FDR <=",MIN_FDR,sep=" "), pch=20)
+                dev.off()  # AVI: added to create separate pdf files
+
+                ## ADD COUNTS TO LOGFILE
+                cat(CompPrefix," genes with FDR <= ",MIN_FDR,": ",nrow(pass.fdr.table)," (up=",nrow(pass.fdr.up.table),", down=",nrow(pass.fdr.down.table),")","\n",file=LogFile,append=TRUE,sep="")
+                cat(CompPrefix," genes with FDR <= ",MIN_FDR," & FC > 2: ",nrow(pass.fdr.logFC.table)," (up=",nrow(pass.fdr.logFC.up.table),", down=",nrow(pass.fdr.logFC.down.table),")","\n",file=LogFile,append=TRUE,sep="")
+
+            }
+            cat("\n",file=LogFile,append=TRUE,sep="")
+        # AVI: creates required output files with message
+        } else {
+            write("More than 2 samples treated with the same antibody are needed to calculate the FDR & LOGFC.", stdout())
+            outfiles <- c(snakemake@output[["FDR_1_perc_res"]],
+                          snakemake@output[["FDR_1_perc_bed"]],
+                          snakemake@output[["FDR_5_perc_res"]],
+                          snakemake@output[["FDR_5_perc_bed"]]
+                          )
+            for (f in outfiles) {
+                writeLines(paste0("More than 2 samples treated with the same antibody are needed to calculate the FDR & LOGFC."), f)
```

Comment:
I think this script should actually fail with this message as an error message in this case. Either users have to provide usable data for this rule or not request the respective output.